windows/src/main/csharp/ch/cyberduck/ui/ViewModels/TransfersViewModel.cs

```diff
@@ -0,0 +1,535 @@
+            OnPropertyChanged();
+            OnPropertyChanged(nameof(ToolbarSeparatorLeft));
+        }
+    }
+
+    public bool ToolbarRemove
+    {
+        get => preferences.getBoolean("transfer.toolbar.remove");
+        set
+        {
+            OnPropertyChanging();
+            preferences.setProperty("transfer.toolbar.remove", value);
+            OnPropertyChanged();
+            OnPropertyChanged(nameof(ToolbarSeparatorLeft));
+        }
+    }
+
+    public bool ToolbarResume
+    {
+        get => preferences.getBoolean("transfer.toolbar.resume");
+        set
+        {
+            OnPropertyChanging();
+            preferences.setProperty("transfer.toolbar.resume", value);
+            OnPropertyChanged();
+            OnPropertyChanged(nameof(ToolbarSeparatorLeft));
+        }
+    }
+
+    public bool ToolbarShow
+    {
+        get => preferences.getBoolean("transfer.toolbar.show");
+        set
+        {
+            OnPropertyChanging();
+            preferences.setProperty("transfer.toolbar.show", value);
+            OnPropertyChanged();
+            OnPropertyChanged(nameof(ToolbarSeparatorRight));
+        }
+    }
+
+    public bool ToolbarStop
+    {
+        get => preferences.getBoolean("transfer.toolbar.stop");
+        set
+        {
+            OnPropertyChanging();
+            preferences.setProperty("transfer.toolbar.stop", value);
+            OnPropertyChanged();
+            OnPropertyChanged(nameof(ToolbarSeparatorLeft));
+        }
+    }
+
+    public bool ToolbarTrash
+    {
+        get => preferences.getBoolean("transfer.toolbar.trash");
+        set
+        {
+            OnPropertyChanging();
+            preferences.setProperty("transfer.toolbar.trash", value);
+            OnPropertyChanged();
+            OnPropertyChanged(nameof(ToolbarSeparatorLeft));
+        }
+    }
+
+    public bool ToolbarSeparatorLeft
+    {
+        get
+        {
+            var left = ToolbarRemove | ToolbarTrash | ToolbarCleanup;
+            var right = ToolbarResume | ToolbarStop | ToolbarReload;
+            return left && right;
+        }
+    }
+
+    public bool ToolbarSeparatorRight
+    {
+        get
+        {
+            return ToolbarOpen | ToolbarShow;
+        }
+    }
+
+    public ReadOnlyObservableCollection<TransferViewModel> Transfers => transfers;
+
+    public TransfersViewModel(
+        TransferController controller,
+        TransfersStore store,
+        Locale locale,
+        BandwidthProvider bandwidthProvider,
+        ConnectionProvider connectionProvider,
+        Preferences preferences,
+        WpfIconProvider iconProvider)
+    {
+        this.controller = controller;
+        this.store = store;
+        this.preferences = preferences;
+        this.iconProvider = iconProvider;
+
+        var localTransfers = store.Transfers.Connect().ObserveOnDispatcher()
+            .Transform(m => new TransferViewModel(controller, m, locale)).DisposeMany()
```

Comment:
Store has dynamic list of TransferModel, which is transformed here to TransferViewModel (on Dispatcher, thus captured SynchronizationContext of SynchronizingObservableObject is set to UI)